bin/yugabyted

```diff
@@ -2283,5 +2481,140 @@ def retry_op(func, timeout=180):
+    def __init__(self):
+        self.ysql_user = os.environ.get('YSQL_USER')
+        self.ysql_password = os.environ.get('YSQL_PASSWORD')
+        self.ysql_db = os.environ.get('YSQL_DB')
+        self.ycql_user = os.environ.get('YCQL_USER')
+        self.ycql_password = os.environ.get('YCQL_PASSWORD')
+        self.ycql_keyspace = os.environ.get('YCQL_KEYSPACE')
+
+    def is_valid(self, var_name):
+        return True if var_name is not None and var_name else False
+
+    def get_ysql_user(self):
+        return self.ysql_user if self.is_valid(self.ysql_user) else DEFAULT_YSQL_USER
+
+    def get_ysql_password(self):
+        return self.ysql_password if self.is_valid(self.ysql_password) else (self.ysql_user if (
+                    self.is_valid(self.ysql_user)) else DEFAULT_YSQL_PASSWORD)
+
+    def get_ysql_db(self):
+        return self.ysql_db if self.is_valid(self.ysql_db) else (self.ysql_user if (
+                    self.is_valid(self.ysql_user)) else DEFAULT_YSQL_DB)
+
+    def get_ycql_user(self):
+        return self.ycql_user if self.is_valid(self.ycql_user) else DEFAULT_YCQL_USER
+
+    def get_ycql_password(self):
+        return self.ycql_password if self.is_valid(self.ycql_password) else (self.ycql_user if (
+                    self.is_valid(self.ycql_user)) else DEFAULT_YCQL_PASSWORD)
+
+    def get_ycql_keyspace(self):
+        return self.ycql_keyspace if self.is_valid(self.ycql_keyspace) else (self.ycql_user if (
+                    self.is_valid(self.ycql_user)) else DEFAULT_YCQL_KEYSPACE)
+
+    def get_ysql_credentials(self, get_default_credentials=False):
+        if get_default_credentials:
+            return DEFAULT_YSQL_USER, DEFAULT_YSQL_PASSWORD, DEFAULT_YSQL_DB
+        else:
+            return self.get_ysql_user(), self.get_ysql_password(), self.get_ysql_db()
+
+    def get_ycql_credentials(self, get_default_credentials=False):
+        if get_default_credentials:
+            return DEFAULT_YCQL_USER, DEFAULT_YCQL_PASSWORD, DEFAULT_YCQL_KEYSPACE
+        else:
+            return self.get_ycql_user(), self.get_ycql_password(), self.get_ycql_keyspace()
+
+    def is_exists(self, var_to_check):
+        return self.is_valid(os.environ.get(var_to_check))
+
+    def setup_ysql_credentials(self, proxy_class):
+
+        # Create DB
+        if self.get_ysql_db() != DEFAULT_YSQL_DB and not proxy_class.db_exists(self.get_ysql_db()):
+            Output.log("Creating {} database".format(self.get_ysql_db()))
+            _, err = proxy_class.create_db(self.get_ysql_db())
+            if err:
+                Output.log_error_and_exit("Failed to create {} database: \
+                    {}".format(self.get_ysql_db(), err))
+
+            Output.log("{} database created successfully".format(self.get_ysql_db()))
+
+        # Update password for default user
+        if self.get_ysql_user() == DEFAULT_YSQL_USER:
+            _, err = proxy_class.update_password(self.get_ysql_password())
```

Comment:
i would make these methods throw exceptions of some sort and avoid the detailed logging before and after each call.

Reply:
I've enabled the `log_cmd` for most of the commands except `create_user` and `update_password` because the `log_cmd` prints the whole command so it'll print the password for both the commands in the logs file.